pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -266,3 +266,7 @@ def on_evaluation_epoch_end(self) -> None:
         self.trainer.call_hook(hook_name)
         self.trainer.call_hook("on_epoch_end")
         self.trainer.logger_connector.on_epoch_end()
+
+    def teardown(self) -> None:
+        self._results.cpu()
+        self.epoch_loop.teardown()
```

Comment:
we are calling teardown again on the epoch_loop? It was already called in run, right?
shouldn't hurt but we may want to be aware of how many times it is getting called.